modal/cli/run.py

```diff
@@ -157,24 +157,27 @@ def _get_click_command_for_function(app: App, function_tag):
     @click.pass_context
     def f(ctx, **kwargs):
         show_progress: bool = ctx.obj["show_progress"]
-        with enable_output(show_progress):
-            with run_app(
-                app,
-                detach=ctx.obj["detach"],
-                environment_name=ctx.obj["env"],
-                interactive=ctx.obj["interactive"],
-            ):
-                if cls is None:
-                    function.remote(**kwargs)
-                else:
-                    # unpool class and method arguments
-                    # TODO(erikbern): this code is a bit hacky
-                    cls_kwargs = {k: kwargs[k] for k in cls_signature}
-                    fun_kwargs = {k: kwargs[k] for k in fun_signature}
-
-                    instance = cls(**cls_kwargs)
-                    method: Function = getattr(instance, method_name)
-                    method.remote(**fun_kwargs)
+        try:
+            with enable_output(show_progress):
+                with run_app(
+                    app,
+                    detach=ctx.obj["detach"],
+                    environment_name=ctx.obj["env"],
+                    interactive=ctx.obj["interactive"],
+                ):
+                    if cls is None:
+                        function.remote(**kwargs)
+                    else:
+                        # unpool class and method arguments
+                        # TODO(erikbern): this code is a bit hacky
+                        cls_kwargs = {k: kwargs[k] for k in cls_signature}
+                        fun_kwargs = {k: kwargs[k] for k in fun_signature}
+
+                        instance = cls(**cls_kwargs)
+                        method: Function = getattr(instance, method_name)
+                        method.remote(**fun_kwargs)
+        except KeyboardInterrupt:
+            pass  # handled within run_app
```

Comment:
This comment confuses me a bit — if the error is handed in the (inner) context manager, why are we catching it here?

Reply:
The error is "handled" by printing stuff and disconnecting/cleaning up the app etc., but we catch the error here in the CLI layer to prevent click from receiving the KeyboardInterrupt since that makes it print its own "Aborted." message